ci/run-test-labshell.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env labshell
+#!/usr/bin/env bash
+#set -o xtrace
+
+ROOT_DIR=$(cd "$(dirname "$0")/.." && pwd)
+
+HW="$(find "$ROOT_DIR/" -mindepth 1 -maxdepth 1 -type d -name 'hw*' | sort | tail -n 1 | sed -nE 's/.*\/hw([0-9]+)$/\1/p')"
+HW_FOLDER="${ROOT_DIR}/hw${HW}"
+HW_SCRIPT="${HW_FOLDER}/hw${HW}.sh"
+echo " found latest hw (${HW}) in '${HW_FOLDER}' with scripts '${HW_SCRIPT}'"
+
+# skip hw < 1
+if [ -z "$HW" ] || [ "$HW" -lt 1 ]; then
+  echo '+ No tests for hw < 1, skipping'
+  exit 0
+fi
+
+# kill qemu
+if pgrep qemu-system-; then
+  echo '! qemu is running, please stop qemu before running this test.'
+  exit 1
+fi
+
+function run() {
+  echo " run '$@'..."
+  if ! "$@"; then
+    echo "! '$@' failed."
+    exit -1
+  fi
+}
+
+# cd into hw folder
+cd "${HW_FOLDER}"
+
+# build
+echo " test 'Build all artifacts' function..."
+run "${HW_SCRIPT}"
+run "${ROOT_DIR}/ci/check-files.py" "${ROOT_DIR}/ci/artifacts/hw${HW}.txt"
+
+# run qemu
+echo " test 'qemu*' script functions..."
+if [ "$HW" -eq 1 ]; then
+  run "${HW_SCRIPT}" qemu_sysinfo &
+  sleep 30
+  echo " verify qemu is running..."
+  run pgrep qemu-system-x86
+  run pkill qemu-system-x86
+  
+  run "${HW_SCRIPT}" qemu_busybox &
+  sleep 30
+  echo " verify qemu is running..."
+  run pgrep qemu-system-x86
+else
+  run "${HW_SCRIPT}" qemu &
+  sleep 30
```

Comment:
How about something like this?

```bash
...
    echo " verify qemu is running..."
    for s in `seq 0 30`; do
        sleep 1
        if [ "$HW" -lt 3 ] && [ run pgrep qemu-system-x86]; then
            break
        elseif [ run pgrep qemu-system-aar ]; then
            break
        elseif [ $s -eq 30 ]; then
            echo Timeout after $s seconds waiting for qemu.
            exit 1
        else
            echo Not running after $s seconds.
        fi
    done
...
```

**Untested code warning!** 💣

Reply:
That looks like a nice enhancement for `ci/run-test-labshell.sh `, but this PR is about the new wrapper script `ci/run-test.sh `.

(Note: Move of `ci/run-test.sh ` to `ci/run-test-labshell.sh ` is a unrecognized by git.)

Reply:
Alright! May I ask you to follow up with this enhancement in another PR? :-)